binary_transparency/firmware/common/signature.go

```diff
@@ -0,0 +1,135 @@
+// Copyright 2020 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"crypto"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha512"
+	"crypto/x509"
+	"encoding/pem"
+	"io/ioutil"
+	"strings"
+
+	"github.com/golang/glog"
+)
+
+func getPrivateKey() (*rsa.PrivateKey, error) {
+	mKey := strings.NewReader(RSApri)
+	priv, err := ioutil.ReadAll(mKey)
+	if err != nil {
+		glog.Exitf("Read failed! %s", err)
+	}
+
+	privPem, _ := pem.Decode(priv)
+	var privPemBytes []byte
+	if privPem.Type != "RSA PRIVATE KEY" {
+		glog.Exitf("RSA private key is of the wrong type %s", privPem.Type)
+	}
+	privPemBytes = privPem.Bytes
+
+	var parsedKey interface{}
+	if parsedKey, err = x509.ParsePKCS1PrivateKey(privPemBytes); err != nil {
+		if parsedKey, err = x509.ParsePKCS8PrivateKey(privPemBytes); err != nil {
+			glog.Exitf("Unable to parse RSA private key %v", err)
+		}
+	}
+
```

Comment:
For the demo, maybe we don't really need to probe different formats since you made the key yourself just now., but if you do think it's useful and want to probe here, then should probably support EC too.